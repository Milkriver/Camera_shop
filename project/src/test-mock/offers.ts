export const mockOffers = [{
    id: 1,
    name: 'Name',
    vendorCode: 'VendorCode',
    type: 'type',
    category: 'category',
    description: 'description',
    level: 'level',
    rating: 5,
    price: 5000,
    previewImg: '',
    previewImg2x: '',
    previewImgWebp: '',
    previewImgWebp2x: '',
    reviewCount: 2,
  }];

  export const mockPromo = {
    id: 1,
    name: '',
    previewImg: '',
    previewImg2x: '',
    previewImgWebp: '',
    previewImgWebp2x: '',
  };

  export const mockOffer = {
    id: 1,
    name: 'Name',
    vendorCode: 'VendorCode',
    type: 'type',
    category: 'category',
    description: 'description',
    level: 'level',
    rating: 5,
    price: 5000,
    previewImg: '',
    previewImg2x: '',
    previewImgWebp: '',
    previewImgWebp2x: '',
    reviewCount: 2,
  };
