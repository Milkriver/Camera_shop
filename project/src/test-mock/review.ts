export const mockReviews = [{
  cameraId: 1,
  userName: '',
  advantage: '',
  disadvantage: '',
  review: '',
  rating: 1,
}];

export const mockReview = {
  id: '',
  userName: '',
  advantage: '',
  disadvantage: '',
  review: '',
  rating: 1,
  createAt: '',
  cameraId: 1,
};

