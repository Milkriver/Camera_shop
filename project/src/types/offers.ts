export type TOfferItem = {
    id: number;
    name: string;
    vendorCode: string;
    type: string;
    category: string;
    description: string;
    level: string;
    rating: number;
    price: number;
    previewImg: string;
    previewImg2x: string;
    previewImgWebp: string;
    previewImgWebp2x: string;
    reviewCount: number;
  }

export type TReview = {
    id: string;
    userName: string;
    advantage: string;
    disadvantage: string;
    review: string;
    rating: number;
    createAt: string;
    cameraId: number;
  }

export type TPromoOffer = {
    id: number;
    name: string;
    previewImg: string;
    previewImg2x: string;
    previewImgWebp: string;
    previewImgWebp2x: string;
  }

export type TReviewPost = {
  cameraId: number;
  userName: string;
  advantage: string;
  disadvantage: string;
  review: string;
  rating: number;
}
export type TCouponPost = {
    coupon: string;
}

export type TOrderPost = {
    camerasIds: number[];
    coupon: string | null;
  }

export type TOrderPosition = {
    id: string;
    totalCount: number;
    totalPrice: number;
    item: TOfferItem;
 }

export type TUpdatedItem = {
  item: TOfferItem;
  newCount: number;
}
