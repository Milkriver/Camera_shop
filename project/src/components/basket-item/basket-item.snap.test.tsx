import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { mockPosition } from '../../test-mock/offers';
import HistoryRouter from '../history-router/history-router';
import BasketItem from './basket-item';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

const history = createMemoryHistory();


describe('Component: BasketItem', () => {
  it('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
      OFFERS: {
        promoOffer: {}
      }
    });
    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <BasketItem product={mockPosition[0]}/>
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
