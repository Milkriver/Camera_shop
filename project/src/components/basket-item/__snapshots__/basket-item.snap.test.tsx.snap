// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BasketItem should render correctly 1`] = `
<div>
  <li
    class="basket-item"
  >
    <div
      class="basket-item__img"
    >
      <picture>
        <source
          srcset="/, /"
          type="image/webp"
        />
        <img
          alt="Name"
          height="120"
          src="/"
          srcset="/"
          width="140"
        />
      </picture>
    </div>
    <div
      class="basket-item__description"
    >
      <p
        class="basket-item__title"
      >
        Name
      </p>
      <ul
        class="basket-item__list"
      >
        <li
          class="basket-item__list-item"
        >
          <span
            class="basket-item__article"
          >
            Артикул: 
          </span>
          <span
            class="basket-item__number"
          >
            VendorCode
          </span>
        </li>
        <li
          class="basket-item__list-item"
        >
          type фотокамера
        </li>
        <li
          class="basket-item__list-item"
        >
          level уровень
        </li>
      </ul>
    </div>
    <p
      class="basket-item__price"
    >
      <span
        class="visually-hidden"
      >
        Цена:
      </span>
      5000
       ₽
    </p>
    <div
      class="quantity"
    >
      <button
        aria-label="уменьшить количество товара"
        class="btn-icon btn-icon--prev"
      >
        <svg
          aria-hidden="true"
          height="12"
          width="7"
        >
          <use
            xlink:href="#icon-arrow"
          />
        </svg>
      </button>
      <label
        class="visually-hidden"
        for="counter1"
      />
      <input
        aria-label="количество товара"
        id="counter1"
        max="99"
        min="1"
        type="number"
        value="1"
      />
      <button
        aria-label="увеличить количество товара"
        class="btn-icon btn-icon--next"
      >
        <svg
          aria-hidden="true"
          height="12"
          width="7"
        >
          <use
            xlink:href="#icon-arrow"
          />
        </svg>
      </button>
    </div>
    <div
      class="basket-item__total-price"
    >
      <span
        class="visually-hidden"
      >
        Общая цена:
      </span>
      5000
       ₽
    </div>
    <button
      aria-label="Удалить товар"
      class="cross-btn"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="10"
        width="10"
      >
        <use
          xlink:href="#icon-close"
        />
      </svg>
    </button>
  </li>
</div>
`;
